src/binding/c/rma_api.txt

```diff
@@ -0,0 +1,466 @@
+        concurrent accumulate calls to the same target address will use the same
+        operation. If set to same_op_no_op, then the implementation will assume that
+        all concurrent accumulate calls to the same target address will use the same
+        operation or 'MPI_NO_OP'. This can eliminate the need to protect access for
+        certain operation types where the hardware can guarantee atomicity. The default
+        is same_op_no_op.
+*/
+
+MPI_Win_create_dynamic:
+    .desc: Create an MPI Window object for one-sided communication
+    .seealso: MPI_Win_attach MPI_Win_detach MPI_Win_allocate MPI_Win_allocate_shared MPI_Win_create MPI_Win_free
+    .impl: mpid
+/*
+    Notes:
+    
+    Users are cautioned that displacement arithmetic can overflow in variables of
+    type 'MPI_Aint' and result in unexpected values on some platforms. This issue may
+    be addressed in a future version of MPI.
+    
+    Memory in this window may not be used as the target of one-sided accesses in
+    this window until it is attached using the function 'MPI_Win_attach'. That is, in
+    addition to using 'MPI_Win_create_dynamic' to create an MPI window, the user must
+    use 'MPI_Win_attach' before any local memory may be the target of an MPI RMA
+    operation. Only memory that is currently accessible may be attached.
+*/
+
+MPI_Win_detach:
+    .desc: Detach memory from a dynamic window
+    .seealso: MPI_Win_create_dynamic MPI_Win_attach
+    .impl: mpid
+/*
+    Notes:
+    Memory also becomes detached when the associated dynamic memory window is freed.
+*/
+{ -- early_return --
+    if (base == NULL)
+        goto fn_exit;
+}
+
+MPI_Win_fence:
+    .desc: Perform an MPI fence synchronization on a MPI window
+    .skip: validate-ASSERT
+    .impl: mpid
+{ -- error_check --
+    CHECKMASK: assert, assert, MPI_MODE_NOSTORE MPI_MODE_NOPUT MPI_MODE_NOPRECEDE MPI_MODE_NOSUCCEED
+}
+
+MPI_Win_flush:
+    .desc: Complete all outstanding RMA operations at the given target
+    .seealso: MPI_Win_flush_all MPI_Win_flush_local MPI_Win_flush_local_all MPI_Win_lock MPI_Win_lock_all
+    .impl: mpid
+    .earlyreturn: pt2pt_proc_null
+
+MPI_Win_flush_all:
+    .desc: Complete all outstanding RMA operations at all targets
+    .seealso: MPI_Win_flush MPI_Win_flush_local MPI_Win_flush_local_all MPI_Win_lock MPI_Win_lock_all
+    .impl: mpid
+
+MPI_Win_flush_local:
+    .desc: Complete locally all outstanding RMA operations at the given target
+    .seealso: MPI_Win_flush MPI_Win_flush_all MPI_Win_flush_local_all MPI_Win_lock MPI_Win_lock_all
+    .impl: mpid
+    .earlyreturn: pt2pt_proc_null
+
+MPI_Win_flush_local_all:
+    .desc: Complete locally all outstanding RMA operations at all targets
+    .seealso: MPI_Win_flush MPI_Win_flush_all MPI_Win_flush_local MPI_Win_lock MPI_Win_lock_all
+    .impl: mpid
+
+MPI_Win_free:
+    .desc: Free an MPI RMA window
+{
+    if (MPIR_Process.attr_free && win_ptr->attributes) {
```

Comment:
Note below is lost in `win_free.c`
```
   Notes:
   If successfully freed, 'win' is set to 'MPI_WIN_NULL'.
```

Reply:
Note is lost in `win_get_group.c` too.

Reply:
In `win_get_group.c`, do we want to replace ` *group = group_ptr->handle;` with `MPIR_OBJ_PUBLISH_HANDLE`?

Reply:
> 
> 
> Note is lost in `win_get_group.c` too.

Added

Reply:
> 
> 
> In `win_get_group.c`, do we want to replace ` *group = group_ptr->handle;` with `MPIR_OBJ_PUBLISH_HANDLE`?

Why not?